pkgs/applications/editors/emacs/make-emacs.nix

```
@@ -460,10 +460,13 @@ mkDerivation (finalAttrs: {
       NATIVE_FULL_AOT = "1";
       LIBRARY_PATH = lib.concatStringsSep ":" libGccJitLibraryPaths;
     }
-    // lib.optionalAttrs (variant == "macport") {
-      # Fixes intermittent segfaults when compiled with LLVM >= 7.0.
-      # See https://github.com/NixOS/nixpkgs/issues/127902
-      NIX_CFLAGS_COMPILE = "-include ${./macport_noescape_noop.h}";
+    // {
+      NIX_CFLAGS_COMPILE = lib.concatStrings [
+        # Fixes intermittent segfaults when compiled with LLVM >= 7.0.
+        # See https://github.com/NixOS/nixpkgs/issues/127902
+        (lib.optionalString (variant == "macport") "-include ${./macport_noescape_noop.h}")
+        (lib.optionalString stdenv.hostPlatform.isDarwin " -DFD_SETSIZE=10000 -DDARWIN_UNLIMITED_SELECT")
```

If the above suggestion is applied, this space is not necessary.
```suggestion
        (lib.optionalString stdenv.hostPlatform.isDarwin "-DFD_SETSIZE=10000 -DDARWIN_UNLIMITED_SELECT")
```

---

Updated as suggested!